src/KubernetesClient/Kubernetes.Generic.cs

```diff
@@ -0,0 +1,1131 @@
+            }
+            if (limit != null)
+            {
+                queryParameters.Add($"limit={System.Uri.EscapeDataString(SafeJsonConvert.SerializeObject(limit, SerializationSettings).Trim('"'))}");
+            }
+            if (resourceVersion != null)
+            {
+                queryParameters.Add($"resourceVersion={System.Uri.EscapeDataString(resourceVersion)}");
+            }
+            if (timeoutSeconds != null)
+            {
+                queryParameters.Add($"timeoutSeconds={System.Uri.EscapeDataString(SafeJsonConvert.SerializeObject(timeoutSeconds, SerializationSettings).Trim('"'))}");
+            }
+            if (watch != null)
+            {
+                queryParameters.Add($"watch={System.Uri.EscapeDataString(SafeJsonConvert.SerializeObject(watch, SerializationSettings).Trim('"'))}");
+            }
+            if (pretty != null)
+            {
+                queryParameters.Add($"pretty={System.Uri.EscapeDataString(pretty)}");
+            }
+            if (queryParameters.Count > 0)
+            {
+                url += "?" + string.Join("&", queryParameters);
+            }
+            // Create HTTP transport objects
+            var httpRequest = new HttpRequestMessage();
+            HttpResponseMessage httpResponse = null;
+            httpRequest.Method = new HttpMethod("GET");
+            httpRequest.RequestUri = new System.Uri(url);
+            // Set Headers
+
+
+            if (customHeaders != null)
+            {
+                foreach(var header in customHeaders)
+                {
+                    if (httpRequest.Headers.Contains(header.Key))
+                    {
+                        httpRequest.Headers.Remove(header.Key);
+                    }
+                    httpRequest.Headers.TryAddWithoutValidation(header.Key, header.Value);
+                }
+            }
+
+            // Serialize Request
+            string requestContent = null;
+            // Set Credentials
+            if (Credentials != null)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                await Credentials.ProcessHttpRequestAsync(httpRequest, cancellationToken).ConfigureAwait(false);
+            }
+            // Send Request
+            if (shouldTrace)
+            {
+                ServiceClientTracing.SendRequest(invocationId, httpRequest);
+            }
+            cancellationToken.ThrowIfCancellationRequested();
+            httpResponse = await HttpClient.SendAsync(httpRequest, cancellationToken).ConfigureAwait(false);
+            if (shouldTrace)
+            {
+                ServiceClientTracing.ReceiveResponse(invocationId, httpResponse);
+            }
+            var statusCode = httpResponse.StatusCode;
+            cancellationToken.ThrowIfCancellationRequested();
+            string responseContent = null;
+            if ((int)statusCode != 200 && (int)statusCode != 401)
```

Comment:
This is inconsistent with the `OK` below, please use constants everywhere for consistency.